urllib3/connection.py

```diff
@@ -316,6 +318,9 @@ def connect(self):
             server_hostname=hostname,
             ssl_context=context)
 
+        if self.transport_security_manager is not None:
+            self.transport_security_manager.validate_connection(self.sock)
```

Comment:
I'm a little bit nervous about using the same method at two very different places in connection setup and management. Can we consider having two methods, with different names?